lightning/pkg/importer/table_import.go

```diff
@@ -765,18 +774,26 @@ ChunkLoop:
 			setError(err)
 			break
 		}
+
+		var memoryUsage int
+		if chunk.FileMeta.Type == mydump.SourceTypeParquet {
+			chunkSize, err := getChunkCompressedSizeForParquet(ctx, chunk, rc.store)
+			memoryUsage = int(chunkSize * 2)
+			if err != nil {
+				setError(err)
+				break
+			}
+			memLimiter.Acquire(memoryUsage) // parquet reader uses 2 as the concurrent reader number
```

Comment:
can we control reader count and memory usage of a single parquet parser within the memory share of a cpu core? so we can still have concurrency for chunk processors = cpu core.

Reply:
I think these code is something like "Defensive Programming". 
Currently, the memory usage of parquet reader is mainly influenced by [data page size](https://parquet.apache.org/docs/file-format/configurations/), but it seems like this value can only be obtained during the parsing process. So it is hard to set a proper value to avoid OOM for massive parquet files.